src/Button/ButtonBase.tsx

```diff
@@ -4,29 +4,59 @@ import Box from '../Box'
 import {merge, SxProp} from '../sx'
 import {useTheme} from '../ThemeProvider'
 import {ButtonProps, StyledButton} from './types'
-import {getVariantStyles, getSizeStyles, getButtonStyles} from './styles'
+import {getVariantStyles, getButtonStyles, getAlignContentSize} from './styles'
+import CounterLabel from '../CounterLabel'
 
 const ButtonBase = forwardRef<HTMLElement, ButtonProps>(
   ({children, as: Component = 'button', sx: sxProp = {}, ...props}, forwardedRef): JSX.Element => {
-    const {leadingIcon: LeadingIcon, trailingIcon: TrailingIcon, variant = 'default', size = 'medium'} = props
+    const {
+      leadingVisual: LeadingVisual,
+      trailingVisual: TrailingVisual,
+      trailingAction: TrailingAction,
+      trailingVisualCount: trailingVisualCount,
+      variant = 'default',
+      size = 'medium',
+      alignContent = 'center',
+      block = false
+    } = props
     const {theme} = useTheme()
     const iconWrapStyles = {
-      display: 'inline-block'
+      display: 'flex',
+      pointerEvents: 'none'
     }
     const sxStyles = useMemo(() => {
-      return merge.all([getButtonStyles(theme), getSizeStyles(size, variant, false), getVariantStyles(variant, theme)])
-    }, [theme, size, variant])
+      return merge.all([getButtonStyles(theme), getVariantStyles(variant, theme)])
+    }, [theme, variant])
+
     return (
-      <StyledButton as={Component} sx={merge(sxStyles, sxProp as SxProp)} {...props} ref={forwardedRef}>
-        {LeadingIcon && (
-          <Box as="span" data-component="leadingIcon" sx={iconWrapStyles}>
-            <LeadingIcon />
-          </Box>
-        )}
-        {children && <span data-component="text">{children}</span>}
-        {TrailingIcon && (
-          <Box as="span" data-component="trailingIcon" sx={{...iconWrapStyles, ml: 2}}>
-            <TrailingIcon />
+      <StyledButton
+        as={Component}
+        sx={merge(sxStyles, sxProp as SxProp)}
+        {...props}
+        ref={forwardedRef}
+        data-component={block ? 'block' : null}
+        data-size={size === 'small' ? 'small' : size === 'large' ? 'large' : undefined}
```

Comment:
We don't want to have `data-size` attribute for medium sizes?

Not a blocker, just a suggestion to maybe improve readability - feel free to ignore if it is not helpful.

If we don't mind having `data-size` attribute for medium sizes maybe we can do `data-size={size}`

If we only want the attribute for `small` and `large` sizes maybe we can do `data-size={size === 'small' || size === 'large' ? size : undefined}`
 Let me know what you think 👀

Reply:
Since medium is default, we don't _really_ need an attribute for it for CSS. I like your second suggestion though!